src/Compilers/CSharp/Test/Semantic/Semantics/FunctionPointerTests.cs

```diff
@@ -1918,5 +1918,34 @@ void M(delegate*<void> ptr)
                 Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "ToString").WithArguments("delegate*<void>", "ToString").WithLocation(6, 13)
             );
         }
+
+        [Fact]
+        public void ArglistCannotBeUsedWithFunctionPointers()
+        {
+            var comp = CreateCompilationWithFunctionPointers(@"
+unsafe class C
+{
+    static void M(delegate*<string, int, void> ptr1, delegate*<__arglist, void> ptr2)
+    {
+        ptr1(__arglist(string.Empty, 1), 1);
```

Comment:
>ptr1(__arglist(string.Empty, 1), 1); [](start = 8, length = 36)

Consider testing when __arglist is the last in the argument list, including the case when it is after all parameters that method can take.